apps/app/src/app/pages/(blocks)/blocks/sidebar/(sidebar).page.ts

```diff
@@ -0,0 +1,56 @@
+import type { RouteMeta } from '@analogjs/router';
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { RouterLink } from '@angular/router';
+import { NgIcon, provideIcons } from '@ng-icons/core';
+import { lucideGithub } from '@ng-icons/lucide';
+import { metaWith } from '@spartan-ng/app/app/shared/meta/meta.util';
+import { SidebarPreview } from '@spartan-ng/app/app/shared/sidebar-preview/sidebar-preview';
+import { HlmButtonImports } from '@spartan-ng/helm/button';
+import { hlmCode } from '@spartan-ng/helm/typography';
+
+export const routeMeta: RouteMeta = {
+	meta: metaWith('spartan/blocks - Sidebar', 'Sidebar blocks using spartan/ui primitives'),
+	title: 'spartan/blocks - Sidebar',
+};
+
+@Component({
+	selector: 'spartan-sidebar',
+	imports: [SidebarPreview, RouterLink, HlmButtonImports, NgIcon],
+	providers: [provideIcons({ lucideGithub })],
+	template: `
+		<div id="sidebar-1" class="flex flex-col gap-4">
+			<div class="flex items-center justify-between">
+				<a
+					routerLink="."
+					fragment="sidebar-1"
+					class="flex-1 text-center text-sm font-medium underline-offset-2 hover:underline md:flex-auto md:text-left"
+				>
+					A sidebar with a sticky header
+				</a>
+				<a
+					hlmBtn
+					size="sm"
+					href="https://github.com/spartan-ng/spartan/blob/main/apps/app/src/app/pages/(sidebar-preview)/sidebar-preview/sidebar-sticky-header.page.ts"
+					target="_blank"
+				>
+					Open in
+					<ng-icon name="lucideGithub" />
+				</a>
+			</div>
+
+			<p class="text-muted-foreground text-sm">
+				Setup a CSS variable for the header height by adding
+				<code class="${hlmCode}">[--header-height:--spacing(14)]</code>
+				to the body or page container and use
+				<code class="${hlmCode}">h-(--header-height)</code>
+				to set the height of your header. This variable is then used to offset the sidebar to account for the sticky
+				header height.
+			</p>
+
+			<!-- TODO iframe width is overflowing the container cutting of content -->
```

Comment:
@MerlinMoos do you have an idea on how to restrict the iframe content to the container width? The example is overflowing and should stay with in the container.

<img width="3048" height="1068" alt="CleanShot 2025-11-07 at 20 05 22@2x" src="https://github.com/user-attachments/assets/f0af4530-abd8-4f94-abb7-9e7131d1d51c" />